### PR TITLE
result set clean up + fix mssql edge case

### PR DESCRIPTION
**Pull request by edublancas**
## Describe your changes

- cleaning up ResultSet
- fixes an edge case when using SQL server
- updates compatibility table
- adds missing integration tests for bar and pie plots

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--781.org.readthedocs.build/en/781/

### CODE REVIEW

File: doc/integrations/compatibility.md [hunk 1]
```diff
@@ -70,20 +90,24 @@ These table reflects the compatibility status of JupySQL `>=0.7`
 
 - Running queries with `%%sql` ✅
 - CTEs with `%%sql --save NAME` ✅
-- Plotting with `%%sqlplot boxplot` ❌
-- Plotting with `%%sqlplot histogram` ❌
-- Plotting with `ggplot` API ❌
+- Plotting with `%%sqlplot boxplot` ✅
+- Plotting with `%%sqlplot bar` ✅
+- Plotting with `%%sqlplot pie` ✅
+- Plotting with `%%sqlplot histogram` ✅
```
Comment (neelasha23) [inline doc/integrations/compatibility.md:96]: Getting error on `taxi` dataset for histogram:

Screenshot 2023-08-04 at 3 27 43 PM

**Comment by neelasha23** [comment]
Running plots on taxi dataset on datetime field is a bit slow (tried on MSSQL) :

```%%time
%sqlplot bar --table taxi --column passenger_count
```
This runs in 555ms:
```
CPU times: user 98 ms, sys: 38.1 ms, total: 136 ms
Wall time: 555 ms
```

Whereas on the `tpep_pickup_datetime` it takes 37.3secs:

Screenshot 2023-08-04 at 3 41 01 PM

Piechart is stuck even after running for a minute:

Screenshot 2023-08-04 at 3 37 36 PM

Should I open another issue for this? @edublancas

File: doc/integrations/compatibility.md (hunk 1 above)
Comment (edublancas) [reply doc/integrations/compatibility.md]: fixed

**Comment by edublancas** [comment]
@neelasha23 yes, please open an issue. I think the problem is that histogram requires counting unique values and since those are timestamp, they're probably all unique. I don't think there's much to do here. perhaps we can show a warning depending on the data type (i.e., if the column they pass is a timestamp, we show the warning)
